File: src/main/java/me/zeroeightsix/kami/module/modules/movement/TimerSpeed.java
package me.zeroeightsix.kami.module.modules.movement;

import me.zeroeightsix.kami.module.Module;
import me.zeroeightsix.kami.module.Module.Category;
import me.zeroeightsix.kami.module.Module.Info;
import me.zeroeightsix.kami.setting.Setting;
import me.zeroeightsix.kami.setting.Settings;
import me.zeroeightsix.kami.util.InfoCalculator;

/**
 * @author TBM
 * Updated by S-B99 on 28/01/20
 */
@Info(name = "TimerSpeed", description = "Automatically change Timer Speed", category = Category.MOVEMENT)
public class TimerSpeed extends Module {
    private float tickDelay = 0.0f;
    private static float curSpeed = 0.0f;
    private Setting<Float> minimumSpeed = register(Settings.floatBuilder("Minimum Speed").withMinimum(0.0F).withMaximum(10.0F).withValue(4.0F));
    private Setting<Float> maxSpeed = register(Settings.floatBuilder("Max Speed").withMinimum(0.0F).withMaximum(10.0F).withValue(7.0F));
    private Setting<Float> attemptSpeed = register(Settings.floatBuilder("Attempt Speed").withMinimum(1.0F).withMaximum(10.0F).withValue(4.2F));
    private Setting<Float> fastSpeed = register(Settings.floatBuilder("Fast Speed").withMinimum(1.0F).withMaximum(10.0F).withValue(5.0F));

    public static String returnGui() {
        return "" + InfoCalculator.round(curSpeed, 2);
    }

    public void onUpdate() {
        if (tickDelay == minimumSpeed.getValue()) {
            curSpeed = fastSpeed.getValue();
            mc.timer.tickLength = 50.0F / fastSpeed.getValue();
        }

        if (tickDelay >= maxSpeed.getValue()) {
            tickDelay = 0;
            curSpeed = attemptSpeed.getValue();
            mc.timer.tickLength = 50.0F / attemptSpeed.getValue();
        }
        ++tickDelay;
    }
    public void onDisable() {
        mc.timer.tickLength = 50.0F;
    }

}

File: src/main/java/me/zeroeightsix/kami/module/modules/movement/TimerSpeed.kt
package me.zeroeightsix.kami.module.modules.movement

import me.zeroeightsix.kami.module.Module
import me.zeroeightsix.kami.setting.Settings

/**
 * @author TBM
 * Updated by dominikaaaa on 28/01/20
 */
@Module.Info(
        name = "TimerSpeed",
        description = "Automatically change your timer to go fast",
        category = Module.Category.MOVEMENT
)
class TimerSpeed : Module() {

    private val minimumSpeed = register(Settings.floatBuilder("MinimumSpeed").withMinimum(0.1f).withMaximum(10.0f).withValue(4.0f).build())
    private val maxSpeed = register(Settings.floatBuilder("MaxSpeed").withMinimum(0.1f).withMaximum(10.0f).withValue(7.0f).build())
    private val attemptSpeed = register(Settings.floatBuilder("AttemptSpeed").withMinimum(1.0f).withMaximum(10.0f).withValue(4.2f).build())
    private val fastSpeed = register(Settings.floatBuilder("FastSpeed").withMinimum(1.0f).withMaximum(10.0f).withValue(5.0f).build())

    private var tickDelay = 0.0f
    private var curSpeed = 0.0f

    override fun onUpdate() {
        if (tickDelay == minimumSpeed.value) {
            curSpeed = fastSpeed.value
            mc.timer.tickLength = 50.0f / fastSpeed.value
        }
        if (tickDelay >= maxSpeed.value) {
            tickDelay = 0f
            curSpeed = attemptSpeed.value
            mc.timer.tickLength = 50.0f / attemptSpeed.value
        }
        ++tickDelay
    }

    override fun onDisable() {
        mc.timer.tickLength = 50.0f
    }
}